frappe/public/js/frappe/views/reports/query_report.js

```diff
@@ -836,16 +836,27 @@ frappe.views.QueryReport = class QueryReport extends frappe.views.BaseList {
 		const custom_format = this.report_settings.html_format || null;
 		const filters_html = this.get_filters_html_for_print();
 		const landscape = print_settings.orientation == 'Landscape';
+		let columns = [];
+
+		if (print_settings.columns) {
```

Comment:
Commonify the column getting logic which is used in `pdf_report` method as well.